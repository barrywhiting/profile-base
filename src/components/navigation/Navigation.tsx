import React, { FC, ReactElement } from 'react';
import * as styles from '../../styles/components/navigation/navigation.module.scss';
import logo from '../../images/barry-profile.svg';

const Navigation: FC = (): ReactElement => {
  return (
      <nav className={styles.nav}>
        <a href='/home'><img src={logo} alt="Barry's Profile Logo" width='110' height='100' /></a>
        <ul>
          <li><a href='/home'>Home</a></li>
          <li><a href='/components'>Components</a></li>
          <li><a href='/styles'>Styles</a></li>
          <li><a href='types'>Types</a></li>
        </ul>
      </nav>);
}

export default Navigation;
