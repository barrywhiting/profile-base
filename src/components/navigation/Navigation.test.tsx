import { render, screen } from "@testing-library/react";

import Navigation from "./Navigation";
import React from "react";

test("renders navigation", () => {
  render(<Navigation />);
  const linkElement = screen.getByText(/Home/i);
  expect(linkElement).toBeInTheDocument();
});

it('renders snapshot', () => {
  const wrapper = render(<Navigation />);
  expect(wrapper).toMatchSnapshot();
});
