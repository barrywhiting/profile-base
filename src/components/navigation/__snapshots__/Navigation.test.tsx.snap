// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <a
          href="/home"
        >
          <img
            alt="Barry's Profile Logo"
            height="100"
            src="test-file-stub"
            width="110"
          />
        </a>
        <ul>
          <li>
            <a
              href="/home"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/components"
            >
              Components
            </a>
          </li>
          <li>
            <a
              href="/styles"
            >
              Styles
            </a>
          </li>
          <li>
            <a
              href="types"
            >
              Types
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <a
        href="/home"
      >
        <img
          alt="Barry's Profile Logo"
          height="100"
          src="test-file-stub"
          width="110"
        />
      </a>
      <ul>
        <li>
          <a
            href="/home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/components"
          >
            Components
          </a>
        </li>
        <li>
          <a
            href="/styles"
          >
            Styles
          </a>
        </li>
        <li>
          <a
            href="types"
          >
            Types
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
